### DISCUSSION TITLE
use binance error

### DISCUSSION THREAD

**Discussion by intcpu** [PrivateRookie/ws-tool#43] [Q&A] [answered]
unresolved imports `ws_tool::codec::AsyncStringCodec`, `ws_tool::connector::async_tcp_connect`, `ws_tool::connector::async_wrap_rustls`

 codec::AsyncStringCodec,
  |     ^^^^^^^----------------
  |     |      |
  |     |      help: a similar name exists in the module: `StringCodec`
  |     no `AsyncStringCodec` in `codec`
6 |     connector::{async_tcp_connect, async_wrap_rustls, get_host},
  |                 ^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^ no `async_wrap_rustls` in `connector`
  |                 |
  |                 no `async_tcp_connect` in `connector`

**Comment by PrivateRookie** [comment]
async features is not enabled by default, try `cargo run --example binance -F async` if you want to run example, or
```toml
ws-tool = { version = "*", features = ["async"] }
```
if you import ws-tool in your package

**Reply by intcpu** [reply]
"Loading ws-tool = { path = "ws-tool", features = ["async"] } resolved some errors, but there is still one remaining:
connector::{async_tcp_connect, async_wrap_rustls, get_host},
  |                                    ^^^^^^^^^^^^^^^^^ no `async_wrap_rustls` in `connector`

**Reply by PrivateRookie** [reply, marked as answer]
tls requires additonal features `async_tls_rustls` or `async_tls_native` , if you want async_wrap_rustls function , please add `async_tls_rustls`

**Reply by intcpu** [reply]
thinks, Following your approach, use ws-tool = { path = "ws-tool", features = ["async", "async_tls_rustls"] } is ok
